stacktrace.go

```diff
@@ -271,41 +286,60 @@ func extractFrames(pcs []uintptr) []Frame {
 	return frames
 }
 
-// filterFrames filters out stack frames that are not meant to be reported to
-// Sentry. Those are frames internal to the SDK or Go.
-func filterFrames(frames []Frame) []Frame {
+// createFrames creates Frame objects while filtering out frames that are not
+// meant to be reported to Sentry, those are frames internal to the SDK or Go.
+func createFrames(frames []runtime.Frame) []Frame {
 	if len(frames) == 0 {
 		return nil
 	}
 
-	// reuse
-	filteredFrames := frames[:0]
+	result := make([]Frame, 0, len(frames))
 
 	for _, frame := range frames {
-		// Skip Go internal frames.
-		if frame.Module == "runtime" || frame.Module == "testing" {
-			continue
+		function := frame.Function
+		var pkg string
+		if function != "" {
+			pkg, function = splitQualifiedFunctionName(function)
 		}
-		// Skip Sentry internal frames, except for frames in _test packages (for
-		// testing).
-		if strings.HasPrefix(frame.Module, "github.com/getsentry/sentry-go") &&
-			!strings.HasSuffix(frame.Module, "_test") {
-			continue
+
+		if !shouldSkipFrame(pkg) {
+			result = append(result, newFrame(pkg, function, frame.File, frame.Line))
 		}
-		filteredFrames = append(filteredFrames, frame)
 	}
 
-	return filteredFrames
+	return result
 }
 
-func isInAppFrame(frame Frame) bool {
-	if strings.HasPrefix(frame.AbsPath, build.Default.GOROOT) ||
+// TODO ID: why do we want to do this?
+// I'm not aware of other SDKs skipping all Sentry frames, regardless of their position in the stactrace.
+// For example, in the .NET SDK, only the first frames are skipped until the call to the SDK.
+// As is, this will also hide any intermediate frames in the stack and make debugging issues harder.
```

Comment:
One example I can think of quickly is that when an error occurs in the BeforeSend callback (or anywhere down the road), than that stacktrace may be confusing as all the sentry-go stacks would be stripped. 
I haven't actually tested this, it just looked weird at a glance when I saw this and wanted to note this may be an issue (as I've said, compared to other SDKs I've seen).

Reply:
Fair case, let's leave the note. Could you also open an issue for this please?

Reply:
Thanks 👍